_posts/2019-11-01-deploying-debezium-with-kafkaconnector-resource.md

```diff
@@ -0,0 +1,697 @@
+---
+layout: post
+title:  "Deploying Debezium using the new KafkaConnector resource"
+date: 2019-11-01
+author: tom_bentley
+---
+
+For too long our Kafka Connect story hasn't quite as "Kubernetes-native" as it could have been.
+We had a `KafkaConnect` resource to configure a Kafka Connect _cluster_ but you still had to use the Kafka Connect REST API to actually create a _connector_ within it.
+While this wasn't especially difficult using something like `curl`, it stood out because everything else could be done using `kubectl` and it meant that connectors didn't fit into our Kubernetes-native vision.
+With the help of a contribution from the community, Strimzi now supports a `KafkaConnector` custom resource and the rest of this blog post is going to explain how to use it using [Debezium](https://debezium.io) as an example. 
+
+And as if that wasn't enough, there's some awesome ASCII art.
+
+<!--more-->
+
+# Debezi-what?
+
+If you've not heard of Debezium before it is an open source project for applying [Change Data Capture](https://en.wikipedia.org/wiki/Change_data_capture) (CDC) pattern to your applications using Kafka.
+
+But what is CDC? You probable have several databases in your organization; silos full of business-critical data.
+While you can query those databases any time you want, the essence of your business revolves around how that data gets modified.
+Those modifications trigger, and are triggered by, real world business events (e.g. a phone call from a customer, or a successful payment or whatever)
+In order to reflect this, modern application architectures are frequently event-based. 
+And so asking for the _current state_ of some tables is not enough; what these architectures need is a stream of _events representing modifications_ to those tables.
+That's what CDC is: Capturing the changes to the state data as event data.
+
+Concretely, Debezium works with a number of common DBMSs (MySQL, MongoDB, PostgreSQL, Oracle, SQL Server and Cassandra) and runs as a source connector within a Kafka Connect cluster.
+How Debezium works on the database side depends which database it's using.
+For example for MySQL it reads the commit log in order to know what transaction are happening, but for MongoDB it hooks into the native replication mechanism.
+In any case, the changes get represented as JSON events (using a common schema) which are sent to a Kafka.
```

Comment:
JSON is just one particular event serialization, it could also be Avro or yet something else.

s/a Kafka/Kafka/